parser_library/src/analyzer.cpp

```diff
@@ -75,11 +75,11 @@ analyzer::analyzer(const std::string& text, analyzer_options opts)
                 *this,
                 opts.get_preprocessor(
                     [libs = &opts.get_lib_provider(), program = opts.file_loc, &ctx = ctx_](std::string_view library) {
-                        std::optional<utils::path::resource_location> res_loc;
+                        std::optional<utils::resource::resource_location> res_loc;
 
                         auto result = libs->get_library(std::string(library), program, &res_loc);
 
-                        if (res_loc.has_value())
+                        if (res_loc.has_value() && !res_loc.value().get_uri().empty())
```

Comment:
Now that the argument is optional, the `get_library` implementations should be either returning std::nullopt or valid string.